catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/RegistryIdPostIngestPlugin.java

```diff
@@ -0,0 +1,147 @@
+
+    private static final int RETRY_INTERVAL = 30;
+
+    private ScheduledExecutorService executorService;
+
+    private CatalogFramework catalogFramework;
+
+    private FilterBuilder filterBuilder;
+
+    private Set<String> registryIds = ConcurrentHashMap.newKeySet();
+
+    public Set<String> getRegistryIds() {
+        return Collections.unmodifiableSet(registryIds);
+    }
+
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+        if (input != null && input.getCreatedMetacards() != null) {
+            input.getCreatedMetacards()
+                    .stream()
+                    .filter(RegistryUtility::isRegistryMetacard)
+                    .forEach(e -> registryIds.add(RegistryUtility.getRegistryId(e)));
+        }
+        return input;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        return input;
+    }
+
+    @Override
+    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+        if (input != null && input.getDeletedMetacards() != null) {
+            input.getDeletedMetacards()
+                    .stream()
+                    .filter(RegistryUtility::isRegistryMetacard)
+                    .forEach(e -> registryIds.remove(RegistryUtility.getRegistryId(e)));
+        }
+        return input;
+    }
+
+    public void init() {
+        try {
+            List<Metacard> registryMetacards;
+            Filter registryFilter = filterBuilder.attribute(Metacard.TAGS)
+                    .is()
+                    .like()
+                    .text(RegistryConstants.REGISTRY_TAG);
+            QueryImpl query = new QueryImpl(registryFilter);
+            query.setPageSize(1000);
+            QueryRequest request = new QueryRequestImpl(query);
+
+            QueryResponse response = Security.runAsAdminWithException(() -> catalogFramework.query(
+                    request));
+
+            if (response == null) {
+                throw new PluginExecutionException(
+                        "Failed to initialize plugin. Query for registry metacards came back null");
+            }
+
+            registryMetacards = response.getResults()
+                    .stream()
+                    .map(Result::getMetacard)
+                    .collect(Collectors.toList());
+            registryIds.addAll(registryMetacards.stream()
+                    .map(RegistryUtility::getRegistryId)
+                    .filter(e -> !StringUtils.isEmpty(e))
+                    .collect(Collectors.toList()));
+        } catch (PrivilegedActionException | PluginExecutionException e) {
+            LOGGER.debug("Error getting registry metacards. Will try again later");
+            executorService.schedule(this::init, RETRY_INTERVAL, TimeUnit.SECONDS);
+        }
+    }
+
+    public void destroy() {
+        executorService.shutdownNow();
```

Comment:
We have used that example for our destroy methods in several places. Guess i just forgot about it here. Updated with appropriate destroy logic for executor.